internal/sharedcomponent/sharedcomponent.go

```diff
@@ -85,18 +74,66 @@ func (c *Component[V]) Unwrap() V {
 
 // Start starts the underlying component if it never started before.
 func (c *Component[V]) Start(ctx context.Context, host component.Host) error {
-	var err error
-	c.startOnce.Do(func() {
-		// It's important that status for a shared component is reported through its
-		// telemetry settings to keep status in sync and avoid race conditions. This logic duplicates
-		// and takes priority over the automated status reporting that happens in graph, making the
-		// status reporting in graph a no-op.
-		c.telemetry.ReportStatus(component.NewStatusEvent(component.StatusStarting))
-		if err = c.component.Start(ctx, host); err != nil {
-			c.telemetry.ReportStatus(component.NewPermanentErrorEvent(err))
-		}
-	})
-	return err
+	if c.hostWrapper == nil {
+		var err error
+		c.startOnce.Do(func() {
+			c.hostWrapper = &hostWrapper{
+				host:           host,
+				sources:        make([]componentstatus.Reporter, 0),
+				previousEvents: make([]*componentstatus.Event, 0),
+			}
+			statusReporter, isStatusReporter := host.(componentstatus.Reporter)
+			if isStatusReporter {
+				c.hostWrapper.addSource(statusReporter)
+			}
+
+			// It's important that status for a shared component is reported through its
+			// telemetry settings to keep status in sync and avoid race conditions. This logic duplicates
+			// and takes priority over the automated status reporting that happens in graph, making the
+			// status reporting in graph a no-op.
+			c.hostWrapper.Report(componentstatus.NewEvent(componentstatus.StatusStarting))
+			if err = c.component.Start(ctx, c.hostWrapper); err != nil {
+				c.hostWrapper.Report(componentstatus.NewPermanentErrorEvent(err))
+			}
+		})
+		return err
+	}
+	statusReporter, isStatusReporter := host.(componentstatus.Reporter)
+	if isStatusReporter {
+		c.hostWrapper.addSource(statusReporter)
+	}
+	return nil
+}
+
+var _ component.Host = (*hostWrapper)(nil)
+var _ componentstatus.Reporter = (*hostWrapper)(nil)
+
+type hostWrapper struct {
+	host           component.Host
+	sources        []componentstatus.Reporter
+	previousEvents []*componentstatus.Event
+}
+
+func (h *hostWrapper) GetExtensions() map[component.ID]component.Component {
+	return h.host.GetExtensions()
+}
+
+func (h *hostWrapper) Report(e *componentstatus.Event) {
+	// Only remember an event if it will be emitted and it has not been sent already.
+	if len(h.sources) > 0 && !slices.Contains(h.previousEvents, e) {
+		h.previousEvents = append(h.previousEvents, e)
+	}
+
+	for _, s := range h.sources {
+		s.Report(e)
+	}
```

Comment:
Not an actionable comment, but I just wanted to highlight that managing multiple sources would be unnecessary if we improved our component instancing story as suggested here: https://github.com/open-telemetry/opentelemetry-collector/issues/10534. Another thing to mention, but also likely unactionable, is that we have two options for reporting status from within the shared component. One is to treat the shared component as multiple instances, one per pipeline. That is the approach taken here. The other, would be to treat it as a single instance in multiple pipelines. The latter is more correct, but there are technical difficulties in achieving that. Part of the reason is that the service graph treats the shared component as multiple instances.